authentication/ruby/microsoft_kiota_authentication/lib/access_token_provider.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+require 'oauth2'
+require_relative 'extensions/oauth2_ext'
+require_relative 'allowed_hosts_validator'
+require_relative 'contexts/client_credential_context'
+require_relative 'contexts/authorization_code_context'
+require_relative 'contexts/on_behalf_of_context'
+
+module MicrosoftKiotaAuthentication
+  # Access Token Provider class implementation
+  class AccessTokenProvider
```

Comment:
split into 2: the generic code needs to move to the access token provider in abstractions
the code that's specific to oauth2/microsoft identity platform stays here
and rename to azure access token provider